pynopegl-utils/pynopegl_utils/viewer/config.py

```diff
@@ -0,0 +1,179 @@
+#
+# Copyright 2023 Clément Bœsch <u pkh.me>
+# Copyright 2023 Nope Foundry
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import json
+import os
+import os.path as op
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Tuple
+
+from PySide6 import QtCore
+
+RESOLUTIONS = {
+    "144p": 144,
+    "240p": 240,
+    "360p": 360,
+    "480p": 480,
+    "720p": 720,
+    "1080p": 1080,
+    "1440p": 1440,
+    "4k": 2048,
+}
+
+
+@dataclass
+class EncodeProfile:
+    name: str
+    format: str
+    args: List[str]
+
+
+ENCODE_PROFILES = dict(
+    mp4_h264_420=EncodeProfile(
+        name="MP4 / H264 4:2:0",
+        format="mp4",
+        args=["-pix_fmt", "yuv420p"],
```

Comment:
I would explicitly specify the libx264 encoder even though it is the default for mp4 (if libx264 is enabled). I would also specify a good enough (tm) crf.

Reply:
1. Added `-c:v libx264`
2. What value for CRF? 21? 18?
3. faststart added (420 only), with a comment on the rationale